lib/engine/game/g_1856.rb

```diff
@@ -355,6 +385,346 @@ def stock_round
           Step::G1856::BuySellParShares,
         ])
       end
+
+      # Nationalization Methods
+
+      def event_nationalization!
+        @nationalization_trigger ||= @round.active_step.current_entity.owner
+        @log << '-- Event: CGR merger --'
+        corporations_repay_loans
+        @nationalizables = nationalizable_corporations
+        @log << "Merge candidates: #{present_nationalizables(nationalizables)}" if nationalizables.any?
+        # starting with the player who bought the 6 train, go around the table repaying loans
+
+        # player picks order of their companies.
+        # set aside compnanies that do not repay succesfully
+
+        # starting with the player who bought the 6 train, go around the table trading shares
+        # trade all shares
+      end
+
+      def nationalizables
+        @nationalizables ||= []
+      end
+
+      def max_national_shares
+        20
+      end
+
+      def corporations_repay_loans
+        @corporations.each do |corp|
+          next unless corp.floated? && corp.loans.size.positive?
+
+          loans_repaid = [corp.loans.size, (corp.cash / loan_value).to_i].min
+          amount_repaid = loan_value * loans_repaid
+          next unless amount_repaid.positive?
+
+          corp.spend(amount_repaid, @bank)
+          @loans << corp.loans.pop(loans_repaid)
+          @log << "#{corp.name} repays #{format_currency(amount_repaid)} to redeem #{loans_repaid} loans"
+        end
+      end
+
+      def merge_major(major)
+        @national_formed = true
+        @log << "-- #{major.name} merges into #{national.name} --"
+        # Trains are transferred
+        major.trains.clone.each do |t|
```

Comment:
dup

Reply:
Is `dup` generally preferred to `clone`, or just in this case?

Reply:
dup is just a soft copy, clone does all internals which is not necessary

Reply:
OK thanks, I've changed this